File: 1 Array & Hashing/1 Contain Duplicate (Easy)/Solution.py
import re

def containsDuplicate(nums: list[int]) -> bool:
    nums.sort()
    for i in range(len(nums) - 1) :
        if (nums[i] == nums[i + 1]) :
            return True
    return False

if __name__ == "__main__" :
    print(containsDuplicate([1, 2, 3, 4]))

File: 1 Array & Hashing/1 Contain Duplicate (Easy)/Solution.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

bool containsDuplicate(vector<int>& nums) {

    sort(nums.begin(), nums.end());
    
    std::size_t i = 0;
    while(i < nums.size() - 1) {
        if (nums[i] == nums[++i]) {
            return true;
        }
    }

    return false;
}

int main() {
    std::vector<int> vec = {1, 2, 3, 1};
    cout << containsDuplicate(vec);
    return 0;
}

